Add contributing documentation (#319)
iotaledger/iota.py: .github/CONTRIBUTING.md

**Note:** If you find a **Closed** issue that seems similar to what you're experiencing, open a new issue and include a link to the original issue in the body of your new one.

### Submitting A Bug Report

To report a bug, [open a new issue](https://github.com/iotaledger/iota.py/issues/new), and be sure to include as many details as possible, using the template.

**Note:** Minor changes such as fixing a typo can but do not need an open issue.

If you also want to fix the bug, submit a [pull request](#pull-requests) and reference the issue.
</details>

<br>

<details>
<summary>Suggest a new feature :bulb:</summary>
<br>

This section guides you through suggesting a new feature. Following these guidelines helps maintainers and the community collaborate to find the best possible way forward with your suggestion.

### Before suggesting a new feature

**Ensure the feature has not already been suggested** by searching on GitHub under [**Issues**](https://github.com/iotaledger/iota.py/issues).

### Suggesting a new feature

To suggest a new feature, talk to the IOTA community and IOTA Foundation members on [Discord](https://discord.iota.org/).

> If the team members approves your feature, they will create an issue for it.

Reviewer: ```suggestion
If the team members approve your feature, they will create an issue for it.
```

</details>

<br>

<details>
<summary>Build a new feature :hammer:</summary>
<br>

This section guides you through building a new feature. Following these guidelines helps give your feature the best chance of being approved and merged.

### Before building a new feature

Make sure to discuss the feature with the developers on [Discord](https://discord.iota.org/).

Otherwise, your feature may not be approved at all.

### Building a new feature

> To build a new feature, check out a new branch based on the `master` branch, and be sure to consider the following:

Reviewer: ```suggestion
To build a new feature, check out a new branch based on the `develop` branch, and be sure to consider the following:
```

❔ So that the new feature takes into account any changes made since the previous release?

> - If the feature has a public facing API, make sure to document it, using [Sphinx](https://www.sphinx-doc.org/en/master/) code comments

Reviewer: ```suggestion
- If the feature will become part of PyOTA's public interface (i.e., it includes methods that will be invoked by other developers in their applications), make sure to document it, using [Sphinx-style](https://www.sphinx-doc.org/en/master/) code comments.
```

</details>

<br>

<details>
<summary>Contribute to the documentation :black_nib:</summary>
<br>

> The Python client library documentation is hosted on https://docs.iota.org, which is built from content in the [documentation](https://github.com/iotaledger/documentation) repository.

Reviewer: ❔ Just confirming, are we planning to migrate `docs` directory in this project to the documentation repo?

Author: Eventually, yes I think that would be a good idea.

Please see the [guidelines](https://github.com/iotaledger/documentation/CONTRIBUTING.md) on the documentation repository for information on how to contribute to the documentation.
</details>

<br>

<details>
<summary>Pull requests :mega:</summary>
<br>

This section guides you through submitting a pull request (PR). Following these guidelines helps give your PR the best chance of being approved and merged.

### Before submitting a pull request

When creating a pull request, please follow these steps to have your contribution considered by the maintainers:

- A pull request should have only one concern (for example one feature or one bug). If a PR addresses more than one concern, it should be split into two or more PRs.

> - A pull request can be merged only if it references an open issue

Reviewer: ```suggestion
- A pull request can be merged only if it references an open issue.
```

    **Note:** Minor changes such as fixing a typo can but do not need an open issue.

> - All code should be well tested

Reviewer: ```suggestion
- All code should include comprehensive unit tests.
```

### Submitting a pull request

The following is a typical workflow for submitting a new pull request:

1. Fork this repository
2. Create a new branch based on your fork
3. Commit changes and push them to your fork
4. Create a pull request against the `master` branch

If all [status checks](https://help.github.com/articles/about-status-checks/) pass, and the maintainer approves the PR, it will be merged.

> **Note:** Reviewers may ask you to complete additional work, tests, or other changes before your pull request can be approved and merged.

Reviewer: ```suggestion
**Note:** Reviewers may request changes before your pull request can be approved and merged.
```

💭 "ask you to complete additional work" felt a bit too much like being given an assignment to me.

</details>

<br>

<details>
<summary>Code of Conduct :clipboard:</summary>
<br>

> This project and everyone participating in it is governed by the [IOTA Code of Conduct](CODE_OF_CONDUCT.md).

Reviewer: ```suggestion
This project and everyone participating in it are governed by the [IOTA Code of Conduct](CODE_OF_CONDUCT.md).
```
